### ISSUE TITLE
multiple result sets

### ISSUE THREAD

**Issue by tekken22** [Microsoft/sqlopsstudio#799]
- SQL Operations Studio Version: sqlops 0.26.7 (723c7cb3bfd80d87fc8a1141632089cd4808495c, 2018-02-16T21:34:47.034Z)
- OS Version: Windows_NT x64 6.1.7601
- Extensions: none
---

Steps to Reproduce:

1. create sql containing multiple separate select statements, in my case it was 7, make sure only 4th and 5th return results, other should not return anything.
2.  Scroll down through results and when you get to 4th there will be a few seconds delay before 4th grid with results shows up, it creates a false impression that none of the queries have results.
3. After 4th grid with results shows, scroll up back to first result and go back to 4th again, the same issue, it will take a few seconds before 4th grid shows up.

**Comment by tekken22** [comment]
forgot to mention if you scroll up/down fast enough 4th and 5th result will never show up.

**Comment by IsItGreyOrGray** [comment]
I can't reproduce this exact issue, but I'm also having issues with multiple result sets so I'll tack onto this issue:

If I have 7 or fewer result sets I'm seeing results in grid view and the Messages pane shows correct information for rows affected, but once I go over 7 result sets the grid view shows place holders for the result columns and the Messages pane shows correct information for rows affected, but there is only data in the grid view for the first 4 sets of data.

**Comment by anthonydresser** [comment]
@IsItGreyOrGray regarding your issue, we have optimizations that if there are more than a certain amount of result sets (i think it was 8) we only render what is currently on screen, otherwise there are bad performance issues. Sounds like this optimization might not be working properly.

Is it when you scroll the results don't render?

**Comment by IsItGreyOrGray** [comment]
Sorry about the late response... 
If I scroll, should the newly "visible" result sets currently on the screen be rendered? If so, they're not. It looks like once I hit 8 result sets it will only ever show me what was visible on the screen to begin with.

I've been playing around and this is what I see...


**Using 7 select statements:**
I get all result sets

![image](https://user-images.githubusercontent.com/33128934/37842042-b6076c28-2e86-11e8-8830-438b7bef61b4.png)

**Using 8 select statements:**
If I drag the Results pane as small as I can , it will render the result sets visible on screen (2) and the rest will be empty.
![image](https://user-images.githubusercontent.com/33128934/37841841-346de958-2e86-11e8-9c65-013b539a201f.png)

If I drag the Results pane as tall as I can , it will render the result sets visible on screen (5) and the rest will be empty.

![image](https://user-images.githubusercontent.com/33128934/37841965-9135abbc-2e86-11e8-9c8e-7de254a46fbf.png)

**Comment by uberbloke** [comment]
We have lots of stored procedures that return multiple result sets and this bug is making Ops Studio almost unusable for us

The following code will trigger the issue in Version 0.28.6, run it, and the results sets (for me) for 3 - 10 are blank (I can see some results for 3 - 10 being painted as the queries execute, but once the whole batch is complete then the result sets are empty)

select top 100 2, * from  sys.objects; WAITFOR DELAY '00:00:01';
select top 100 3, * from  sys.objects; WAITFOR DELAY '00:00:01';
select top 100 4, * from  sys.objects; WAITFOR DELAY '00:00:01';
select top 100 5, * from  sys.objects; WAITFOR DELAY '00:00:01';
select top 100 6, * from  sys.objects; WAITFOR DELAY '00:00:01';
select top 100 7, * from  sys.objects; WAITFOR DELAY '00:00:01';
select top 100 8, * from  sys.objects; WAITFOR DELAY '00:00:01';
select top 100 9, * from  sys.objects; WAITFOR DELAY '00:00:01';
select top 100 10, * from  sys.objects; WAITFOR DELAY '00:00:01';

**Comment by kevcunnane** [comment]
@anthonydresser this looks very much like the repro for #1118 - are these all duplicate issues? If so what can we do to resolve?

**Comment by kburtram** [comment]
I'm going to close this as a duplicate of https://github.com/Microsoft/sqlopsstudio/issues/156 so we can track this work in one place.  Please leave related feedback in https://github.com/Microsoft/sqlopsstudio/issues/156.
